### PR TITLE
FreeBSD implement process_doesnt_exist

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c [hunk 1]
```diff
@@ -190,7 +191,22 @@ static attach_state_t ptrace_waitpid(pid_t pid) {
...
+#if defined(__FreeBSD__)
+  struct kinfo_proc kproc;
+  size_t klen;
+  int mib[] = { CTL_KERN, KERN_PROC, KERN_PROC_PID, pid };
+  if (sysctl(mib, 4, &kproc, &klen, NULL, 0) == -1) {
+    return true;
+  }
+
+  switch (kproc.ki_stat) {
+    case SSTOP:
```
Comment (battleblow) [inline src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c]: The comment on SSTOP in proc.h says "Process debugging or suspension.".  This seems to indicate that the process exists, it is just suspended.  So I think that it is only SZOMB that indicates that it doesn't exist?

File: src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c [hunk 2]
```diff
@@ -357,7 +373,7 @@ static bool read_lib_info(struct ps_prochandle* ph) {
   for (i = 0; i < cnt; i++) {
     kve = &freep[i];
     if ((kve->kve_flags & KVME_FLAG_COW) &&
-        kve->kve_path != NULL &&
+        kve->kve_path[0] != '\0' &&
```
Comment (battleblow) [inline src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c]: What is the reasoning behind this change?

**Review by battleblow — CHANGES_REQUESTED** [review]

File: src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c (hunk 2 above)
Comment (devnexen) [reply src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c]: kve_path is an array, thus can t be NULL

File: src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c (hunk 1 above)
Comment (devnexen) [reply src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c]: yes good point.

File: src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c (hunk 2 above)
Comment (battleblow) [reply src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c]: Thanks.  Would 'strnlen(kve->kve_path, PATH_MAX) > 0' be more readable?

Comment (devnexen) [reply src/jdk.hotspot.agent/bsd/native/libsaproc/ps_proc.c]: No preference nor opinion here, same to me but will commit nevertheless.

**Pull request merged**